### PR TITLE
feat: smart upsell check

### LINKED ISSUE #772: Optimize conversions: smart upsell check
Check "No, thanks" instead of "Yes, count me in" if the person doesn't have enough in their balance to pay for 2 years.

### PR DESCRIPTION
**Pull request by Marchand-Nicolas**
close: #772

### CODE REVIEW

**Review by fricoben — APPROVED** [review]
Doesn't work it creates several bugs :)

https://www.loom.com/share/cb1eb4e45c5849eeaffaa479ab888e36

**Review by fricoben — CHANGES_REQUESTED** [review]
Miss clicked my previous review it doesn't work https://github.com/starknet-id/app.starknet.id/pull/777#pullrequestreview-2054886919

File: components/domains/steps/upsellCard.tsx [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
   enabled,
   onUpsellChoice,
+  invalidBalance,
+  displayedCurrency,
 }) => {
+  const [isUserChoice, setIsUserChoice] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsUserChoice(false);
+  }, [displayedCurrency]);
```
Comment (fricoben) [inline components/domains/steps/upsellCard.tsx]: Using displayed currency is not in my opinion a good way to do it. 

Imo you should simply say, that by default the discount is on but if `invalidBalance` is true then you put it off. Then you manage the manual change on top of that.

**Review by fricoben — CHANGES_REQUESTED** [review]
Stll doesn't work for me. 

Please TEST before sending the PR, you have different case with different balances to test here.

https://www.loom.com/share/b2b8fa3296a54b6792cb8c1c4f0ebee3

## Cases to test
- [ ] Balance > discount > normal
- [ ] discount > balance > normal
- [ ] discount > normal > balance

**Review by irisdv — APPROVED** [review]
lgtm in terms of code, I'll let @fricoben test it again

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**